### PR TITLE
Update o365-data-locations.md

### PR DESCRIPTION
**Pull request by ninobb**
The Exchange Online URL was leading to a bookmark at the very end of the page where it should land on top of it.

### CODE REVIEW

File: microsoft-365/enterprise/o365-data-locations.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ description: "Determine where your Microsoft 365 customer data is stored worldwi
 
 See the following links to understand workload data location.
 
-- Exchange Online (EXO)  [Exchange Online Data Location](m365-dr-workload-exo.md#how-can-i-determine-customer-data-location)
+- Exchange Online (EXO)  [Exchange Online Data Location](https://learn.microsoft.com/microsoft-365/enterprise/m365-dr-workload-exo)
```
Comment (JohanFreelancer9) [inline microsoft-365/enterprise/o365-data-locations.md]:
```suggestion
- Exchange Online [Data Residency for Exchange Online](/microsoft-365/enterprise/m365-dr-workload-exo)
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - @ninobb, kindly commit the suggested changes. Many thanks.

**Comment by ninobb** [comment]
Sure, looks good to me.

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by yogkumgit** [comment]
@kelleyvice-msft Please merge this PR. Thanks

**Pull request merged**